### PR TITLE
add section about the new demo server

### PR DESCRIPTION
**Pull request by nilsnolde**
we now finalized the fossgis demo server which hosts an app now: https://valhalla.openstreetmap.de. the repo is available here: https://github.com/gis-ops/valhalla-app. 

I guess it has the best reach if we include it fairly top-level in the root README, so I put it right under the LICENSE. I also re-arranged a little to make the build parts more coherent. I don't have strong feelings about the wording or whatever, I'd just like to include that section where it's seen immediately ~~without scrolling~~. ok for everyone?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,6 +27,10 @@ Valhalla is an open source routing engine and accompanying libraries for use wit
 
 Valhalla, and all of the projects under the Valhalla organization, use the [MIT License](COPYING).  Avatar/logo by [Jordan](https://www.instagram.com/jaykaydraws/)
 
+## Demo Server
```
Comment (kevinkreiser) [inline README.md:30]: maybe this should go at least below the overview after we've described a little bit more of what valhalla is.

Comment (nilsnolde) [reply README.md]: agreed, I moved it below the overview

**Comment by TimMcCauley** [comment]
@nilsnolde  @kevinkreiser there is a part at https://github.com/valhalla/valhalla/blob/master/docs/api/turn-by-turn/api-reference.md that has a link to the old demo server ("View an interactive demo"), maybe this could also be replaced?

**Comment by nilsnolde** [comment]
yep, good idea @TimMcCauley , I also added it to the isochrone api-reference, in case that's fine for the others

File: docs/api/turn-by-turn/api-reference.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 Valhalla's routing service (a.k.a. turn-by-turn), is an open-source routing service that lets you integrate routing and navigation into a web or mobile application.
 
-[View an interactive demo](http://valhalla.github.io/demos/routing)
+[View an interactive demo](https://valhalla.openstreetmap.de)
```
Comment (kevinkreiser) [inline docs/api/turn-by-turn/api-reference.md:5]: i dont know im iffy on these types of changes. i think there is room for both types of demos. the programs we have have in the demos repo are for developers who are running their own valhalla and want to play with it. the demo at valhalla.openstreetmap.de is more for high level people who want to just see vaguely how it works. i'm happy to have both linked but i dont want to lose visibility of the tools we have in the demos repo. i personally think linking to the new interactive demos that are hosted makes the most sense on the main README because that is where people who are only taking a shallower read and who would appreciate a front end would tend to go.

basically i dont want to lose the links to the existing demos and tools even if they require you to be running your own service and are terrible as UI goes hahahha, not saying we cant put in the link to the new stuff, just dont want to lose the old stuff as i think it still serves a purpose (debugging for developers)

how can we serve both use-cases?

Comment (nilsnolde) [reply docs/api/turn-by-turn/api-reference.md]: no worries, I think you're right. this is more the area where a dev is roaming around, makes sense to link more of a dev tool. I removed those changes. that reminds me though that the demos could be updated to hit the open server rather than the key-protected current server to work out-of-the-box, right?

**Review by kevinkreiser — APPROVED** [review]
